pyuvdata/uvcal.py

```diff
@@ -0,0 +1,289 @@
+                'ant_2_array, but there may be extras as well.')
+        self._antenna_names = uvp.UVParameter('antenna_names',
+                                              description=desc,
+                                              form=('Nants_telescope',),
+                                              expected_type=str)
+
+        desc = ('List of integer antenna numbers corresponding to antenna_names,'
+                'shape (Nants_telescope). There must be one '
+                'entry here for each unique entry in ant_1_array and '
+                'ant_2_array, but there may be extras as well.')
+        self._antenna_numbers = uvp.UVParameter('antenna_numbers',
+                                                description=desc,
+                                                form=('Nants_telescope',),
+                                                expected_type=int)
+
+        desc = ('Number of antennas in the array. May be larger ' +
+                'than the number of antennas with data')
+        self._Nants_telescope = uvp.UVParameter('Nants_telescope',
+                                                description=desc,
+                                                expected_type=int)
+
+        desc = 'Array of frequencies, shape (Nspws, Nfreqs), units Hz'
+        self._freq_array = uvp.UVParameter('freq_array', description=desc,
+                                           form=('Nspws', 'Nfreqs'),
+                                           expected_type=np.float,
+                                           tols=1e-3)  # mHz
+
+        desc = ('Channel width of of a frequency bin. Units Hz.')
+        self._channel_width = uvp.UVParameter('channel_width',
+                                              description=desc,
+                                              expected_type=np.float,
+                                              tols=1e-3)
+
+        desc = ('Array of polarization integers, shape (Npols). '
+                'AIPS Memo 117 says: stokes 1:4 (I,Q,U,V);  '
+                'circular -1:-4 (RR,LL,RL,LR); linear -5:-8 (XX,YY,XY,YX)')
+        self._polarization_array = uvp.UVParameter('polarization_array',
+                                                   description=desc,
+                                                   expected_type=int,
+                                                   acceptable_vals=list(np.arange(-8, 0)) + list(np.arange(1, 5)),
+                                                   form=('Npols',))
+
+        desc = ('Array of times, center of integration, shape (Ntimes), ' +
+                'units Julian Date')
+        self._time_array = uvp.UVParameter('time_array', description=desc,
+                                           form=('Ntimes',),
+                                           expected_type=np.float,
+                                           tols=1e-3 / (60.0 * 60.0 * 24.0))
+
+        desc = ('Integration time of a time bin. Units in JD')
```

Comment:
This has different units than the UVData integration_time parameter (which has units of seconds). I think it should match UVData.